### ISSUE TITLE
Images URLs issues with WPML

### ISSUE THREAD

**Issue by RealBigB** [timber/timber#405]
WPML overloads the standard wp `home_url` function to add the language either as a path element (ie: `http://www.mydomain.tld/` becomes `http://www.mydomain.tld/fr/`) or as a query string (`http://www.mydomain.tld/` becomes `http://www.mydomain.tld/?lang=fr`). This breaks a whole lot of things in timber-image-helper.php, and potentially some more in at least timber-image.php and timber-url-helper.php. One of the breakage might not be obvious: it makes the `resize` filter re-sideload and re-resize images that are local and already resized. 

I replaced every call to `home_url()` by `site_url()` in my local instal (in the three aforementioned files) and it seems to solve the problem but I can't garantee it doesn't break something else...

**Comment by jarednova** [comment]
Closed by #502

**Comment by jedrzejchalubek** [comment]
It seems that there is still something wrong. After adding thumbnail to the translated version of post it generates new image file with md5 as name (instead of picking already added).

`twig custompost.thumbnail.src|resize(x,y)` returns relative path `/wp-content/uploads/2015/03/7ab1cdd0648fd5faefb0aa4813d7a583.jpg`

When viewing post in main language, everything works perfectly.

**Comment by RealBigB** [comment]
You have to replace _all_ calls to `home_url()` with calls to `site_url()` to solve the issue, cf my first post.

**Comment by jedrzejchalubek** [comment]
This is not a good way to go. For now, I'll try to get around without timber (get_the_post_thumbnail or something).

**Comment by laurentvd** [comment]
I'm also experiencing this issue when using the latest WPML and latest Timber. Additionally, I'm running WordPress from a subdir (`/wp`). The default language works just fine, but the alternative language fails. It generates url's like `http://domain.ltdhttp://domain.ltd/sv/path/to/image`. 

So `home_url()` returns with the suffix of the locale and `site_url()` returns with the `/wp` suffix. Not really sure how this can be resolved for everyone, but I had to solve it using a hard-coded update to `timber-image-helper.php`.

**Comment by jarednova** [comment]
@laurentvd can you post a snippet of the twig code that you're using? I was able to setup PHPUnit with multisite, so I'd like to give it a shot with what you've got

**Comment by laurentvd** [comment]
Sure! 
```
<img class="photo" src="{{ photo.src | resize(428, 428) }}" 
    width="214" height="214" alt="{{ photo.alt }}"/>
```
I guess for most cases it would suffice to replace `home_url()` with `site_url()`. What I did is basically something like this: `https://codex.wordpress.org/Giving_WordPress_Its_Own_Directory`

**Comment by oxyc** [comment]
This still doesn't work with eg `soil` that outputs relative URLs. https://github.com/timber/timber/blob/master/lib/Image.php#L303

**Comment by palmiak** [comment]
I just hit this bug:
- Bedrock
- WPMU
- Latest Timber

`site_url` gave me:
```
http://domain.test/wp/app/uploads/2018/11/e86f1f56675832cc147988191637f392.jpg
instead of
http://domain.test/app/uploads/2018/11/e86f1f56675832cc147988191637f392.jpg
```

I quickfixed it with:
```
add_filter('timber/URLHelper/file_system_to_url', 'image_url_fixer' );

function image_url_fixer( $url ) {
	$url = str_replace( 'wp/', '', $url );
	return $url;
}
```

I have to digg deeper

**Comment by palmiak** [comment]
@oxyc @jarednova I think I got it.

Problem with Soil & WPML looks like the problem with resize (and other operations) using ` analyze_url()` - there is a fragment:
```
if ( !$result['absolute'] ) {
	$tmp = untrailingslashit(network_home_url()).$tmp;
}
```

I changed it to:
```
if ( !$result['absolute'] ) {
	$tmp = untrailingslashit(site_url()).$tmp;
}
```
and it looks ok. 

On the other hand my bug and @laurentvd (the part about https://codex.wordpress.org/Giving_WordPress_Its_Own_Directory ) looks simillar as Bedrock (in my case) uses /wp folder to do this. I must play with it a bit more. Maybe I'll find some universal solution.
